src/plugins/exitscript/exitscript.js

```diff
@@ -0,0 +1,135 @@
+	selector = "." + componentName,
+	initEvent = "wb-init" + selector,
+	$document = wb.doc,
+	exiturlparam = componentName + "-urlparam",
+	i18nDict = {
+		en: {
+			"msgboxHeader": "Warning",
+			"exitMsg": "You are about to leave a secure site, do you wish to continue?",
+			"yesBtn": "Yes",
+			"cancelBtn": "Cancel"
+
+		},
+		fr: {
+			"msgboxHeader": "Avertissement",
+			"exitMsg": "Vous êtes sur le point de quitter un site sécurisé. Voulez-vous continuer?",
+			"yesBtn": "Oui",
+			"cancelBtn": "Annuler"
+
+		}
+	},
+
+	/**
+	 * @method init
+	 * @param {jQuery Event} event Event that triggered the function call
+	 */
+	init = function( event ) {
+		var elm = wb.init( event, componentName, selector ),
+			settings,
+			$elm;
+		if ( elm ) {
+			$elm = $( elm );
+			settings = $.extend(
+				true,
+				window[ componentName ],
+				wb.getData( $elm, componentName )
+
+			);
+
+			$elm.data( componentName, settings );
+
+			i18nDict = i18nDict[ wb.lang || "en" ];
+
+			if ( $elm.hasClass( exiturlparam ) ) {
+				this.outerHTML = "<a href='" + sessionStorage.getItem( "originalURL" ) + "'>" + sessionStorage.getItem( "originalURL" ) + "</a>";
+				if ( document.querySelector( selector + "-targetparam" ) ) {
+					document.querySelector( selector + "-targetparam" ).outerHTML = sessionStorage.getItem( "trgAttribute" );
+				}
+			}
+			wb.ready( $elm, componentName );
+		}
+	};
+
+$document.on( "click", selector, function( event ) {
+	event.preventDefault();
+
+	var elm = event.currentTarget,
+		$elm = $( elm ),
+		moDalId,
+		wrapper,
+		moDal = document.createDocumentFragment(),
+		tpl = document.createElement( "div" ),
+		settings =  $elm.data( componentName ),
+		modalWindow = settings.displayModal !== false,
+		msgboxHeader = settings.msgboxHeader || i18nDict.msgboxHeader,
+		exitMsg = settings.exitMsg || i18nDict.exitMsg,
+		yesBtn = settings.yesBtn || i18nDict.yesBtn,
+		cancelBtn = settings.cancelBtn || i18nDict.cancelBtn,
+		exitURL = settings.exitURL,
+		targetAttribute = this.getAttribute( "target" ) ? "target='" + this.getAttribute( "target" ) + "'" : "",
+		re = new RegExp( "^(http|https)://", "i" );
+
+	// Check browser support and exit url does not begin with http:// or https://
+	if ( !re.test( exitURL ) ) {
+		sessionStorage.setItem( "trgAttribute", this.getAttribute( "target" ) );
+		sessionStorage.setItem( "originalURL", this );
+
+	}
+
+	if ( modalWindow ) {
+		tpl.innerHTML = "<section class='mfp-hide modal-dialog modal-content overlay-def'>" +
+			"<header class='modal-header'><h2 class='modal-title'>" + msgboxHeader + "</h2></header>" +
+			"<div class='modal-body'>" +
+			"<p>" + exitMsg + "</p>" +
+			"<ul class='list-inline text-center'>" +
+			"<li><button class='btn btn-primary popup-modal-dismiss pull-left' id='eCancel'>" + cancelBtn + "</button></li>" +
+			"<li><a class='btn btn-default pull-right popup-modal-close'" + targetAttribute + " href='" + exitURL + "' type='button'>" + yesBtn + "</a></li>" +
```

Comment:
To make the "popup-modal-dismiss" working, it probably need to be defined in the dialog footer

Reply:
@duboisp It's fixed.

Reply:
@duboisp Pierre. I had put back popup-modal-close, popup-modal-dismiss is no working for me.

Reply:
popup-modal-dismiss will not work when target is undefined.

$( document ).on( "click", ".popup-modal-dismiss", function( event ) {
	if ( !this.hasAttribute( "target" ) ) {
		event.preventDefault();
	}

	$.magnificPopup.close();
} );

Reply:
Hi @duboisp,

Here is the markup that you can add in a wet-boew test page.
Please test when you get a chance.
Scenario – 1: The class popup-modal-dismiss is not working for “Yes” button. The popup-modal-dismiss close the modal dialog but not triggered the link 
Scenario – 2: The anchor has target attribute, the “Yes” button is working as expected.

Thank you
------------
`<p>Scenario – 1 <a class="wb-lbx lbx-modal" title="Example of a modal popup" href="#inline_content_modal">Modal popup</a></p>
		<section id="inline_content_modal" class="mfp-hide modal-dialog modal-content overlay-def">
			<header class="modal-header">
				<h2 class="modal-title">Title</h2>
			</header>
			<div class="modal-body">
				<p>You are about to leave a secure site, do you wish to continue?</p>	
				
			</div>
			<div class='modal-footer'>
			<ul class='list-inline text-center'>
			<li><button class='btn btn-primary popup-modal-dismiss pull-left' id='eCancel'>Cancel</button></li>
			<li><a class='btn btn-default pull-right popup-modal-dismiss' href='http://www.google.ca' type='button'>Yes</a></li>
			</ul>
			
		</section>
<p>Scenario – 2 <a class="wb-lbx lbx-modal" target="_blank" title="Example of a modal popup" href="#inline_content_modal-1">Modal popup -2</a></p>
		<section id="inline_content_modal-1" class="mfp-hide modal-dialog modal-content overlay-def">
			<header class="modal-header">
				<h2 class="modal-title">Title</h2>
			</header>
			<div class="modal-body">
				<p>You are about to leave a secure site, do you wish to continue?</p>	
			</div>
			<div class='modal-footer'>
			<ul class='list-inline text-center'>
			<li><button class='btn btn-primary popup-modal-dismiss pull-left' id='eCancel'>Cancel</button></li>
			<li><a class='btn btn-default pull-right popup-modal-dismiss' target="_blank"  href='http://www.google.ca' type='button'>Yes</a></li>
			</ul>
			</div>
		</section>`

Reply:
@duboisp 

```
<p>Scenario – 1 <a class="wb-lbx lbx-modal" title="Example of a modal popup" href="#inline_content_modal">Modal popup</a></p>
		<section id="inline_content_modal" class="mfp-hide modal-dialog modal-content overlay-def">
			<header class="modal-header">
				<h2 class="modal-title">Title</h2>
			</header>
			<div class="modal-body">
				<p>You are about to leave a secure site, do you wish to continue?</p>	
				
			</div>
			<div class="modal-footer">
			<ul class="list-inline text-center">
			<li><button class="btn btn-primary popup-modal-dismiss pull-left" id="eCancel">Cancel</button></li>
			<li><a class="btn btn-default pull-right popup-modal-dismiss" href="http://www.google.ca" type="button">Yes</a></li>
			</ul>
			
		</section>
<p>Scenario – 2 <a class="wb-lbx lbx-modal" target="_blank" title="Example of a modal popup" href="#inline_content_modal-1">Modal popup -2</a></p>
		<section id="inline_content_modal-1" class="mfp-hide modal-dialog modal-content overlay-def">
			<header class="modal-header">
				<h2 class="modal-title">Title</h2>
			</header>
			<div class="modal-body">
				<p>You are about to leave a secure site, do you wish to continue?</p>	
			</div>
			<div class="modal-footer">
			<ul class="list-inline text-center">
			<li><button class="btn btn-primary popup-modal-dismiss pull-left" id="eCancel">Cancel</button></li>
			<li><a class="btn btn-default pull-right popup-modal-dismiss" target="_blank"  href="http://www.google.ca" type="button">Yes</a></li>
			</ul>
			</div>
		</section>
```